src/base/static/utils/geospatial-utils.js

```diff
@@ -0,0 +1,183 @@
+import hash from "object-hash";
+import { featureCollection } from "@turf/helpers";
+
+const getBufferFeature = (placeGeometry, bufferOptions) => {
+  try {
+    return buffer(placeGeometry, bufferOptions.distance, {
+      units: bufferOptions.units,
+    });
+  } catch (e) {
+    // eslint-disable-next-line no-console
+    console.error(e);
+
+    return null;
+  }
+};
+
+export default {
+  pointInPolygon: ({
+    config = { propertiesToPluck: [] },
+    sourceFeatures,
+    placeGeometry,
+  }) => {
+    // Find the first polygon in which the passed point resides.
+    let foundFeature;
+    try {
+      foundFeature = sourceFeatures.find(feature => {
+        return booleanPointInPolygon(placeGeometry, feature);
+      });
+    } catch (e) {
+      // eslint-disable-next-line no-console
+      console.error(e);
+
+      return {};
+    }
+
+    return foundFeature
+      ? config.propertiesToPluck.reduce(
+          (memo, property) => ({
+            ...memo,
+            [property.name]: foundFeature.properties[property.name],
+          }),
+          {},
+        )
+      : config.propertiesToPluck.reduce(
+          (memo, property) => ({
+            ...memo,
+            [property.name]: property.fallbackValue,
+          }),
+          {},
+        );
+  },
+  aggregatePointsInBuffer: ({ placeGeometry, sourceFeatures, config }) => {
+    const bufferFeature = getBufferFeature(placeGeometry, config.buffer);
+
+    if (!bufferFeature) {
+      return {};
+    }
+
+    const pointsWithin = pointsWithinPolygon(
+      featureCollection(sourceFeatures),
+      bufferFeature,
+    );
+
+    if (config.aggregator.type === "totalCount") {
+      return {
+        [config.name]: pointsWithin.features.length,
+      };
+    } else if (config.aggregator.type === "countByUniquePropertyValues") {
+      // Return an object that counts the number of occurrences of each unique
+      // value of `config.aggregator.property` in the set of points in the
+      // buffer, producing an object like:
+      // {
+      //   propertyA: 3,
+      //   propertyB: 7,
+      //   ...
+      // }
+      const uniqueValues = pointsWithin.features.reduce(
+        (memo, feature) => ({
+          ...memo,
+          [feature.properties[config.aggregator.property]]: memo[
+            feature.properties[config.aggregator.property]
+          ]
+            ? ++memo[feature.properties[config.aggregator.property]]
+            : 1,
+        }),
+        {},
+      );
+
+      return {
+        [config.name]: uniqueValues,
+      };
+    } else {
+      return {
+        [config.name]: pointsWithin.features.length,
+      };
+    }
+  },
+  aggregatePolygonsOverlappingBuffer: ({
+    placeGeometry,
+    sourceFeatures,
+    config,
+  }) => {
+    const bufferFeature = getBufferFeature(placeGeometry, config.buffer);
+
+    if (!bufferFeature) {
+      return {};
+    }
+
+    sourceFeatures = sourceFeatures.reduce((memo, feature) => {
```

Comment:
Instead of mutating our state, I would prefer re-assigning this to a new `const` like `flattenedSourceFeatures` or renaming the original arg `rawSourceFeatures`.